### ISSUE TITLE
JSON schema "format" property

### ISSUE THREAD

**Issue by stalniy** [ThomasAribart/json-schema-to-ts#26]
JSONSchema supports `format` on `type = string` and format `date-time` I want `FromSchema` to return `Date` object instead of string.

There may be other custom formats like `object-id` (MongoDB special type), so what I suggest is to allow users to specify mapping of format value to some scalar type:

```ts
type User = FromSchema<typeof schema, {
  format: {
    "date-time": Date,
    "object-id": ObjectId
  }
}>
```

**Comment by stalniy** [comment]
Never mind. I decided to go with custom solution

**Comment by ThomasAribart** [comment]
I still think it could be an interesting feature. 'json-schema-to-ts' would definitely benefit from handling custom deserialization.

However, it would not be a simple road. I reopen the PR add it as a milestone.

**Comment by leoblum** [comment]
+1. Also wants this feature to use it with mongodb.

**Comment by isakstarlander** [comment]
+1. We have quite a few places in our code base that would benefit of proper Date handling. How would you recommend going about this for the time being?

**Comment by ThomasAribart** [comment]
@isakstarlander @leoblum @stalnyi Alright, it seems this feature is the next on the list :) I'll try to implement it by the end of this month.

For the moment, there's not better way than replacing by hand the property type from the parsed result, for instance with `ts-toolbelt` `Update` utility: https://millsp.github.io/ts-toolbelt/modules/object_p_update.html

**Comment by ThomasAribart** [comment]
@stalniy @leoblum @isakstarlander It took some time but the feature is now available in v2.3.0 ✨ (done in https://github.com/ThomasAribart/json-schema-to-ts/pull/64)

See an example in the README: https://github.com/ThomasAribart/json-schema-to-ts#deserialization
